### DISCUSSION TITLE
VueJS Vite - production build modal not loading

### DISCUSSION THREAD

**Discussion by mwilc0x** [Web3Auth/Web3Auth#729] [Help]
Hi all,

I am adding the Web3Auth sign in modal to our existing Vue app that is built with [Vite](https://github.com/vitejs/vite). 

I have it mostly working in the development environment. It can load the module and I can sign in with social and solana wallets.

But when I compile the build for production and run that, it spins and does not load. 

`Development`:

![Screen Shot 2022-10-15 at 6 57 35 AM](https://user-images.githubusercontent.com/161528/195982805-25d34049-6122-4061-985f-4cb44b10b898.png)

`Production`:

![Screen Shot 2022-10-15 at 6 54 43 AM](https://user-images.githubusercontent.com/161528/195982719-9d77dc34-bfe3-4c0f-98fc-5c9af3a25f95.png)

Our code is pretty standard I think. I followed the example Vue code [here](https://github.com/Web3Auth/examples/blob/main/web-modal-sdk/solana/vue-solana-web3auth-example/src/Home.vue).

There were some things I had to work around, like it was asking for NodeJS methods like `require`, `process` and `Buffer`. I am not sure if that is having an effect in the production build. I have not changed much else except that this is running in a Vite build that I think is using Rollup under the hood for bundling.

For those I did some polyfill from the main `index.html`:

```js
<script>window.global = window;</script>
<script type="module">
  import process from "process";
  import { Buffer } from "buffer";

  window.Buffer = Buffer;
  window.process = process;
</script>
```

and in the `vite.config.js`:

```
build: {
  target: 'es2020',
  commonjsOptions: {
    transformMixedEsModules: true
  }
},
```

Also, currently our build is client side only. I was thinking of converting the build to be server side rendered. I don't know if that would solve it.

If anyone has any ideas or has run in to a similar, any suggestion or feedback how can debug is appreciated.

Thank you

**Comment by mwilc0x** [comment]
I found the common issues link with instructions for Vite. [link](https://web3auth.io/docs/troubleshooting/webpack-issues#vitesvelterollup)

Replaced the vite config with this and removed the `window` overrides. It does solve those problems, but there is a `require is not defined`. We are using Solana wallets.

![image](https://user-images.githubusercontent.com/161528/196008564-0bcd877b-2930-468e-a332-0e4775499bd4.png)

This will suppress that `require` warning under `build` in vite config, but I don't know if that is causing the modal loading to stall:
```
commonjsOptions: {
  transformMixedEsModules: true
},
```

This is what our init code looks like:

```
const web3auth = new Web3Auth({
  clientId,
  chainConfig: {
    chainNamespace: CHAIN_NAMESPACES.SOLANA,
    chainId: "0x3",
    rpcTarget: "https://rpc.ankr.com/solana",
  },
});
```

Also, we are currently a client side only build. We want to convert to Nuxt but in our backlog for right now.

**Comment by mwilc0x** [comment]
This is the require that it is breaking on:

https://github.com/torusresearch/solana-embed/blob/5fa5ae37b7e78dafa6ac249209bd6b6d4a2d9fd4/src/embed.ts#L37

I can fix this manually or use that `transformMixedEsModules` above and it will not break on it. But then the Web3Auth modal will still spin on load and does not seem to initiating for the production build. I followed the Vue/Vite polyfill instructions from the docs. The dev build is fine and is working.

**Comment by mwilc0x** [comment]
Here is a hosted version: https://smy.ecotokens.net/login

I added the domains to allowlist

**Comment by mwilc0x** [comment]
Bump

Does anyone have a Vue/Vite prod build working?

**Comment by mwilc0x** [comment]
Friendly bump

**Comment by chaitanyapotti** [comment]
Hey.. we just released new version of Web3Auth 3.0.4 which has the fix

**Comment by chaitanyapotti** [comment]
This is fixed further in 4.4.2
